src/test/java/project/eepw/softwaretestingcrud/domain/user/data/UserServiceTest.java

```diff
@@ -0,0 +1,369 @@
+package project.eepw.softwaretestingcrud.domain.user.data;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import project.eepw.softwaretestingcrud.domain.user.entity.User;
+import project.eepw.softwaretestingcrud.infrastructure.exception.UserNotFoundException;
+
+class UserServiceTest {
+
+  @Mock
+  private UserRepository userRepository;
+  @InjectMocks
+  private UserService userService;
+  private AutoCloseable closeable;
+
+  @BeforeEach
+  void openMocks() {
+    closeable = MockitoAnnotations.openMocks(this);
+  }
+
+  @AfterEach
+  void releaseMocks() throws Exception {
+    closeable.close();
+  }
+
+  @Test
+  void shouldReturnTheUserWhenUserWithThatIdIsInDB() {
+    //given
+    long id = 1L;
+    User user = User.builder()
+        .id(id)
+        .name("user1")
+        .email("user@example.com")
+        .surname("Doe")
+        .password("password")
+        .posts(Collections.emptySet())
+        .build();
```

Comment:
extract all of these to some private static method and return only builder to set ids